presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java

```diff
@@ -78,10 +77,12 @@ public QueryRunner(
         setupHttpProxy(builder, httpProxy);
         setupBasicAuth(builder, session, user, password);
 
-        if (kerberosEnabled) {
+        if (kerberosRemoteServiceName.isPresent()) {
+            checkArgument(session.getServer().getScheme().equalsIgnoreCase("https"),
```

Comment:
please extract adding this check to a separate commit, especially that it requires second thought

Reply:
done

Reply:
Oh, i see we already have such a check for basic auth.